GVFS/GVFS/CommandLine/StatisticsVerb.cs

```diff
@@ -0,0 +1,209 @@
+                string percent = this.FormatPercent(enlistmentStatistics.GetHydrationOfDirectory(topLevelDirectoriesByHydration[i]));
+                this.Output.WriteLine(" " + percent + " | " + dir);
+            }
+
+            bool healthyRepo = (enlistmentStatistics.GetPlaceholderPercentage() + enlistmentStatistics.GetModifiedPathsPercentage()) < (0.5);
+
+            this.Output.WriteLine("\nRepository status: " + (healthyRepo ? "Healthy" : "Unhealthy"));
+        }
+
+        /// <summary>
+        /// Takes a fractional double and formats it as a percent taking exactly 4 characters with no decimals
+        /// </summary>
+        /// <param name="percent">Fractional double to format to a percent</param>
+        /// <returns>A 4 character string formatting the percent correctly</returns>
+        private string FormatPercent(double percent)
+        {
+            return percent.ToString("P0").PadLeft(4);
+        }
+
+        /// <summary>
+        /// Parse a line of the git index coming from the ls-tree endpoint in the git process to get the path to that file
+        /// </summary>
+        /// <param name="line">The line from the output of the git index</param>
+        /// <returns>The path extracted from the provided line of the git index</returns>
+        private string TrimGitIndexLine(string line)
+        {
+            return line.Substring(line.IndexOf("blob") + 46);
+        }
+
+        /// <summary>
+        /// Talk to the mount process across the named pipe to get a list of the modified paths
+        /// </summary>
+        /// <param name="enlistment">The enlistment being operated on</param>
+        /// <returns>An array containing all of the modified paths in string format</returns>
+        private string[] GetModifiedPathsFromPipe(GVFSEnlistment enlistment)
+        {
+            using (NamedPipeClient pipeClient = new NamedPipeClient(enlistment.NamedPipeName))
+            {
+                string[] modifiedPathsList = { };
+
+                if (!pipeClient.Connect())
+                {
+                    this.ReportErrorAndExit("Unable to connect to GVFS.  Try running 'gvfs mount'");
+                }
+
+                try
+                {
+                    NamedPipeMessages.Message modifiedPathsMessage = new NamedPipeMessages.Message(NamedPipeMessages.ModifiedPaths.ListRequest, "1");
+                    pipeClient.SendRequest(modifiedPathsMessage);
+
+                    NamedPipeMessages.Message modifiedPathsResponse = pipeClient.ReadResponse();
+                    if (!modifiedPathsResponse.Header.Equals(NamedPipeMessages.ModifiedPaths.SuccessResult))
+                    {
+                        this.Output.WriteLine("Bad response from modified path pipe: " + modifiedPathsResponse.Header);
+                        return modifiedPathsList;
+                    }
+
+                    modifiedPathsList = modifiedPathsResponse.Body.Split('\0');

```

Comment:
You can pass in `StringSplitOptions.RemoveEmptyEntries` to avoid the `Take` below if that is only to remove an empty entry at the end.

Reply:
OK cool, I didn't know about that feature, thanks!